### ISSUE TITLE
Fix `no-empty-source` false positives for non-standard syntaxes

### ISSUE THREAD

**Issue by kjugi** [stylelint/stylelint#8547]
### What minimal example or steps are needed to reproduce the bug?

```js
import { css } from "@emotion/react";
import styled from "@emotion/styled";

const BASE_DETAILS_STYLES = css`
  margin: 0;
  padding: 0;
`;

export const BaseDetails = styled.details`
  ${BASE_DETAILS_STYLES}
`;
```

### What minimal configuration is needed to reproduce the bug?

stylelint config:

```js
module.exports = {
  cwd: __dirname,
  customSyntax: "postcss-styled-syntax",
  extends: ["stylelint-config-standard"],
};
```

### How did you run Stylelint?

devbox demo available here: https://codesandbox.io/p/devbox/tky59c

running via

```cli
npm run lint:css
```

script is:
```json
{
"scripts": {
  "lint:css": "stylelint **/*.theme.*"
}
}
```

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "16.19.0",
    "stylelint-config-standard": "^38.0.0"
}
```

### What did you expect to happen?

`Unexpected empty source  no-empty-source` is not reported for a valid code

### What actually happened?

Generated output:

```cli
✖  Unexpected empty source  no-empty-source
```

### Do you have a proposal to fix the bug?

It was introduced by this PR: https://github.com/stylelint/stylelint/pull/8536 in the latest version (16.19.0)

**Comment by ybiquitous** [comment]
I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ybiquitous** [comment]
I felt fixing the bug was a bit challenging, so I just created PR #8548 by myself.

**Comment by ybiquitous** [comment]
I just released <https://github.com/stylelint/stylelint/releases/tag/16.19.1> and confirmed this regression bug was successfully fixed on [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICWBbADhATgFwATFzAGcjcBfXAM0whVwB0QABGFCbJCAOwHpMYAhmGyMA3PS6oMOXEWwBPADYwoVGnUYs2HbjzlKVYiRMhc5uAEIBBAMoBRAPoAROwBUrASQAyNhzdcAml52NrgAvIQkAAYSuLgoApgA5khc8LgADOJccWgCUFCpSelZElHZEjAAHtJ4puYWAkQwTjDYAkiKpBH6ylAAdLDtnUQxObgAJMDW9s5unj5+gcE2ZGWiIAA0IJSdMAByAihwiNVHaMr9AFZEVVvg3LtJCCDAsQwg1dgwXFBEjOkANrvOKMXowRSpbAAWlMT2hcgEv0SUEY7wAupt3owwABXOS0GzyLjtKoAj4YOTEIgIhR9BHE0lorhke5wpBJABiWAS2BeN2491gaCILze4zBdIhUPJjAAjAA2fpygCcysYWIlIHBkJJsMeHNpSKgKNlIEUAm+cg12JAlOw1NpBigDJJAjJCA+FqtIhAElZZCAA)! 🎉 

<img width="151" alt="Image" src="https://github.com/user-attachments/assets/3263ea70-3249-474f-bdf5-cbf518dee253" />
<img width="135" alt="Image" src="https://github.com/user-attachments/assets/7c6f6378-ca48-428e-a08d-1b08bda726e3" />

@kjugi Thank you again for the bug report. 😊

**Comment by kjugi** [comment]
Wow, thanks @ybiquitous for taking a look on this and quick fix! 🙇
